### PR TITLE
Adding support for net 7.0

### PR DESCRIPTION
**Pull request by JRahnama**
Since net8.0 is in preview, we have added support for net7.0 to be able to have access to some of new APIs.When Net8 is released net7.0 will be replaced with net8.0 for GA release.

### CODE REVIEW

**Comment by roji** [comment]
@JRahnama nice to see this... Out of curiosity, which .NET 7.0 features are you specifically looking at?

**Comment by JRahnama** [comment]
@roji, the goal is to be able to completely eliminate the SqlTypeWorkarounds.cs file for Net 8 and greater versions, but since Net 8 is still in preview stage we decided to go by Net7.0 for preview releases and when net8 GA version is released replace Net7.0 with Net8.

**Comment by Wraith2** [comment]
once this is merged if you change the net8 to net7 in my PR https://github.com/dotnet/SqlClient/pull/1934 then new paths should light up and remove the unneeded workarounds.

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec
Comment (David-Engel) [inline tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec:34]: NIT
```suggestion
      <group targetFramework="net7.0">
```

File: tools/specs/Microsoft.Data.SqlClient.nuspec
Comment (DavoudEshtehari) [inline tools/specs/Microsoft.Data.SqlClient.nuspec:116]:
```suggestion
      <group targetFramework="net7.0">
```

File: src/Microsoft.Data.SqlClient/netcore/src/.editorconfig
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netcore/src/.editorconfig:15]: I'd rather keep the report in the error list and unblock the build.
```suggestion
dotnet_diagnostic.SYSLIB0039.severity = suggestion
```

**Review by DavoudEshtehari — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Pull request merged**
